src/cheri/cheri-pte-ext.adoc

```diff
@@ -79,45 +83,63 @@ as usual, and capability reads are controlled by the CRG bit.
+ {cheri_excep_name_pte_st}.
+
+When the CW bit is set, the "capability state", then the behaviour of
+capability loads is controlled by CRG and the behaviour of capability stores
+is controlled by CD.
 
 When CW is set, the CRG bit indicates the current generation of the virtual memory page with
 regards to the ongoing capability revocation cycle. Two schemes are permitted:
 
 * A {cheri_excep_name_pte_ld} is raised when a capability load or AMO instruction is executed
-with <<c_perm>> granted and the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG and PTE.U is set.
+ with:
+** <<c_perm>> granted, and
+** the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG, and
+** PTE.U is set.
 * A {cheri_excep_name_pte_ld} is raised when a capability load or AMO instruction is executed
-with <<c_perm>> granted and the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG and PTE.U is set
-and the capability read from memory has its tag set.
+ with:
+** <<c_perm>> granted, and
+** the virtual page's CRG bit does not equal <<sstatusreg_pte,sstatus>>.UCRG, and
+** PTE.U is set, and
+** the capability read from memory has its {ctag} set.
+
+The second option is enabled by the _{cheri_priv_crg_load_tag_ext}_ extension.
+
+When CW is set, the CD bit indicates that a capability was stored to the
+virtual page since the last time the CD bit was cleared.
+When a capability store or AMO instruction is executed, the {ctag} bit
+of the capability being written is set and the CD bit is clear,
+two schemes are permitted (also see <<section_hardware_pte_updates>>):
+
+* The same behavior as when CW is clear, allowing software interpretation
+ of this state.
+* The implementation sets the CD bit. The PTE update must be
+ atomic with respect to other accesses to the PTE, and must atomically check
+ that the PTE is valid and grants sufficient permissions. Updates to the CD bit
+ must be exact (i.e., not speculative), and observed in program order by the
+ local hart. Furthermore, the PTE update must appear in the global memory order
+ no later than the explicit memory access, or any subsequent explicit memory
+ access to that virtual page by the local hart. The ordering on loads and
+ stores provided by FENCE instructions and the acquire/release bits on atomic
+ instructions also orders the PTE updates associated with those loads and
+ stores as observed by remote harts.
+ The PTE update is not required to be atomic with respect to the explicit memory
+ access that caused the update, and the sequence is interruptible. However, the
+ hart must not perform explicit memory accesses before the PTE update is globally
+ visible.
+
+NOTE: _{cheri_priv_crg_ext}_ requires that when CW, CD and CRG are all clear,
+the implementation clears loaded tags and raises {cheri_excep_name_pte_st} when
+the stored {ctag} is set. Other bit combinations when CW=0 are reserved
+for future extensions.
```

Comment:
the table doesn't show that - it shows PTE.CW=0 and PTE.CD=0
I think you need a row for PTE.CW=0 and PTE.CD=0 separately and if you hit the reserved case what happens? what exception type?